pkg/util/versionchecker/BUILD.bazel

```diff
@@ -0,0 +1,104 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+# Clone empty version of cert-manager repo and list all tags
+genrule(
+    name = "git_tags",
+    outs = [":git_tags.txt"],
+    cmd = "git ls-remote -t --refs https://github.com/jetstack/cert-manager.git | awk '{print $$2;}' | sed 's/refs\\/tags\\///' | sed -n '/v1.0.0-alpha.0/,$$p' > $@",
+)
+
+genrule(
+    name = "test_manifests",
+    srcs = [
+        ":git_tags.txt",
+        "//deploy/manifests:cert-manager.yaml",
+        "//:version",
+    ],
+    outs = ["test_manifests.tar"],
+    cmd = """
+CURRENT_VERSION=$$(cat $(location //:version))
+cp $(location //deploy/manifests:cert-manager.yaml) "$$CURRENT_VERSION.yaml"
+for tag in $$(cat $(location :git_tags.txt))
+do
+    # The "v1.2.0-alpha.1" manifest contains duplicate crds, skip for tests
+    if [[ $$tag == "v1.2.0-alpha.1" ]]; then
+        continue
+    fi
+
+    {
+        HTTP_CODE=$$(curl --compressed -sLo "$$tag.yaml" --write-out "%{http_code}" https://github.com/jetstack/cert-manager/releases/download/$$tag/cert-manager.yaml)
+        if [[ $${HTTP_CODE} -lt 200 || $${HTTP_CODE} -gt 299 ]]; then
+            (mv "$$tag.yaml" "$$tag.notfound")
+        fi
+    } &
```

Comment:
I think these scripts ought to be extracted into `pkg/util/versionchecker/testdata/fetch.sh` so that the tests can be run by non-bazel users.

https://dave.cheney.net/2016/05/10/test-fixtures-in-go

I also wondered if the it'd be better to pass multiple URLs to curl rather than launch all these curl sub-processes, which will grow as the number of tags grows.


```
man curl

URL
       The URL syntax is protocol-dependent. You'll find a detailed description in RFC 3986.

       You can specify multiple URLs or parts of URLs by writing part sets within braces and quoting the URL as in:

         "http://site.{one,two,three}.com"

       or you can get sequences of alphanumeric series by using [] as in:

         "ftp://ftp.example.com/file[1-100].txt"

         "ftp://ftp.example.com/file[001-100].txt"    (with leading zeros)

         "ftp://ftp.example.com/file[a-z].txt"

       Nested sequences are not supported, but you can use several ones next to each other:

         "http://example.com/archive[1996-1999]/vol[1-4]/part{a,b,c}.html"

       You can specify any amount of URLs on the command line. They will be fetched in a sequential manner in the specified order. You can specify command line options and URLs mixed and in any order on the command line.

       You can specify a step counter for the ranges to get every Nth number or letter:

         "http://example.com/file[1-100:10].txt"

         "http://example.com/file[a-z:2].txt"

       When  using  [] or {} sequences when invoked from a command line prompt, you probably have to put the full URL within double quotes to avoid the shell from interfering with it. This also goes for other characters treated spe‐
       cial, like for example '&', '?' and '*'.

       Provide the IPv6 zone index in the URL with an escaped percentage sign and the interface name. Like in

         "http://[fe80::3%25eth0]/"

       If you specify URL without protocol:// prefix, curl will attempt to guess what protocol you might want. It will then default to HTTP but try other protocols based on often-used host name prefixes. For example, for host  names
       starting with "ftp." curl will assume you want to speak FTP.

       curl will do its best to use what you pass to it as a URL. It is not trying to validate it as a syntactically correct URL by any means but is instead very liberal with what it accepts.

       curl will attempt to re-use connections for multiple file transfers, so that getting many files from the same server will not do multiple connects / handshakes. This improves speed. Of course this is only done on files speci‐
       fied on a single command line and cannot be used between separate curl invokes.
```

Reply:
> This will continue if one or more of the required test yamls fails to download, but this will mean that people or CI might end up not running the full test suite.
> Wouldn't it be better to fail the build if any of these downloads fails?
> 
> ```
> $ man curl
> 
>       -f, --fail
>               (HTTP) Fail silently (no output at all) on server errors. This is mostly done to enable scripts etc to better deal with failed attempts. In normal cases when an HTTP server fails to deliver a document,  it  returns  an
>               HTML document stating so (which often also describes why and more). This flag will prevent curl from outputting that and return error 22.
> 
>               This method is not fail-safe and there are occasions where non-successful response codes will slip through, especially when authentication is involved (response codes 401 and 407).
> 
>               See also --fail-with-body.
> ```

This migth cause problems when a new version tag is added, but the manifest is not yet uploaded to git. @wallrj what do you think?

Reply:
The release manifests are uploaded before the tag is created, I think.
The git tag gets added by GitHub when the draft release created by `cmrel` is finally published.
So I don't think this will be a problem.

https://github.com/cert-manager/release/blob/2078202305ef9f246bc1997d75befb07dff71e3f/cmd/cmrel/cmd/gcb_publish.go#L291-L343